### PR TITLE
fix #17615(runnableExamples silently ignored if placed after some code)

### LINKED ISSUE #17615: runnableExamples silently ignored if placed after some code
### Example
```nim
func big*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".} =
  ## Constructor for `JsBigInt`.
  when nimvm: doAssert false, "JsBigInt can not be used at compile-time nor static context" else: discard
  runnableExamples:
    doAssert big(1234567890) == big"1234567890"
    doAssert 0b1111100111.big == 0o1747.big and 0o1747.big == 999.big
```

### Current Output
see https://nim-lang.github.io/Nim/jsbigints.html#big%2CSomeInteger

image

### Expected Output
CT error or support it properly (fixing jsbigints is not enough; docgen needs to be fixed)

### Additional Information
1.5.1 f0587e02e255b1fc1a0d0b21aebcacf37ec43c8f

### PR DESCRIPTION
**Pull request by ringabout**
close #17615

### CODE REVIEW

**Comment by xflywind** [comment]
Whether the comments behind code block error or warning should be considered.

**Comment by ringabout** [comment]
Yeah, I have no experience about runnableExamples tester :) Will revisit the tests tomorrow.

File: tests/nimdoc/t17615.nim [hunk 1]
```diff
@@ -0,0 +1,12 @@
+discard """
+  cmd: "nim doc -r $file"
+  errormsg: "Runnable Examples cannot stay behind the actual code"
+  line: 10
+"""
+
+func big*(integer: int): int =
```
Comment (timotheecour) [inline tests/nimdoc/t17615.nim]: minimized bugs in regression tests are better (my bad, i should've minimized https://github.com/nim-lang/Nim/issues/17615 when i reported the issue)

File: compiler/docgen.nim [hunk 2]
```diff
@@ -607,20 +607,24 @@ proc getAllRunnableExamplesImpl(d: PDoc; n: PNode, dest: var Rope, state: Runnab
...
+        let id = $d.listingCounter
+        dest.add(d.config.getOrDefault"doc.listing_start" % [id, "langNim", ""])
+        var dest2 = ""
+        renderNimCode(dest2, code, isLatex = d.conf.cmd == cmdRst2tex)
+        dest.add dest2
+        dest.add(d.config.getOrDefault"doc.listing_end" % id)
+        return rsRunnable
+      else:
+        localError(d.conf, n.info,
+            errUser, "Runnable Examples cannot stay behind the actual code")
```
Comment (timotheecour) [inline compiler/docgen.nim]:
```suggestion
            errUser, "runnableExamples must appear before the first non-comment statement")
```

(`runnableExamples` is more searchable than Runnable Examples)

**Comment by timotheecour** [comment]
@xflywind i took liberty to push new commits to your PR, hope you don't mind
> Yeah, I have no experience about runnableExamples tester :) Will revisit the tests tomorrow.

```
nim r nimdoc/tester.nim
# if succeeds, you're done, else:
# inspect the diff (expected vs gotten) both manually and visually with rendered pages in a browser
# if all looks good then:
nim r -d:fixup nimdoc/tester.nim # this will error again
nim r nimdoc/tester.nim # this should now succeed
```

then i committed and ran `git push xflywind HEAD:b51` to push on this PR directly

I also addressed the other comments i raised above

## note
if instructions in `nimdoc/tester.nim` aren't clear enough, PR welcome to improve them (other people got confused by it too)

## EDIT
non-controversial bugfix, merging.

**Review by timotheecour — APPROVED** [review]

**Pull request merged**